### ISSUE TITLE
Sorald sometimes adds imports when it really shouldn't

### ISSUE THREAD

**Issue by slarse** [SpoonLabs/sorald#382]
I'm trying to do another PR for verificatum/verificatum-vmn (commit 2e32b06), but am getting some really strange imports added:

```
$ sorald repair --stats-output-file 2142.json --rule-keys 2142 --original-files-path . --file-output-strategy IN_PLACE
```
This results in the following diff:

```diff
diff --git a/src/java/com/verificatum/protocol/hvzk/CCPoSW.java b/src/java/com/verificatum/protocol/hvzk/CCPoSW.java
index afe386d..1b40b7f 100644
--- a/src/java/com/verificatum/protocol/hvzk/CCPoSW.java
+++ b/src/java/com/verificatum/protocol/hvzk/CCPoSW.java
@@ -152,6 +152,7 @@ public final class CCPoSW extends ProtocolElGamal implements CCPoS {
             try {
                 commitmentExportThread.join();
             } catch (final InterruptedException ie) {
+                Thread.currentThread().interrupt();
             }
         }
     }
@@ -256,6 +257,7 @@ public final class CCPoSW extends ProtocolElGamal implements CCPoS {
             try {
                 commitmentExportThread.join();
             } catch (final InterruptedException ie) {
+                Thread.currentThread().interrupt();
             }
         }
 
diff --git a/src/java/com/verificatum/protocol/mixnet/ShufflerElGamalSession.java b/src/java/com/verificatum/protocol/mixnet/ShufflerElGamalSession.java
index 05aa99d..48183b3 100644
--- a/src/java/com/verificatum/protocol/mixnet/ShufflerElGamalSession.java
+++ b/src/java/com/verificatum/protocol/mixnet/ShufflerElGamalSession.java
@@ -25,6 +25,8 @@
  */
 
 package com.verificatum.protocol.mixnet;
+import static com.verificatum.protocol.elgamal.ProtocolElGamal.getCiphPGroup;
+import static com.verificatum.protocol.elgamal.ProtocolElGamal.getWidePublicKey;
 
 import java.io.File;
 
@@ -939,6 +941,7 @@ public final class ShufflerElGamalSession extends ProtocolElGamal {
                 nextOutputThread.join();
             } catch (final InterruptedException ie) {
                 throw new ProtocolError("Unable to join threads!", ie);
+                Thread.currentThread().interrupt();
             }
             nextOutputThread = null;
         }
```

Note the `import static` toward the bottom of the diff. Not only are those imports unrelated to the patch, they're also not supposed to be there as the methods in question are inherited from `ProtocolElGamal`. Needless to say, there's something wrong with how Sorald figures out what to import.

**Comment by slarse** [comment]
This is a bug in the import cleaner in Spoon. Working on a fix.

**Comment by slarse** [comment]
Fixed with #3798, just need the next beta. It can wait until Monday.

**Comment by slarse** [comment]
This was fixed with the new beta version of Spoon!
